src/mobu/services/business/nublado.py

```diff
@@ -157,57 +158,36 @@ async def execute_code(self, session: JupyterLabSession) -> None:
     async def close(self) -> None:
         await self._client.close()
 
-    def annotations(self) -> dict[str, str]:
-        """Timer annotations to use.
-
-        Subclasses should override this to add more annotations based on
-        current business state.  They should call ``super().annotations()``
-        and then add things to the resulting dictionary.
-        """
-        result = {}
-        if self._image:
-            result["image"] = (
-                self._image.description
-                or self._image.reference
-                or "<image unknown>"
-            )
-        if self._node:
-            result["node"] = self._node
-        return result
-
     async def startup(self) -> None:
-        if self.options.jitter:
-            with self.timings.start("pre_login_delay"):
-                max_delay = self.options.jitter.total_seconds()
-                delay = self._random.uniform(0, max_delay)
-                if not await self.pause(timedelta(seconds=delay)):
-                    return
-        await self.hub_login()
-        if not await self._client.is_lab_stopped():
-            try:
-                await self.delete_lab()
-            except JupyterTimeoutError:
-                msg = "Unable to delete pre-existing lab, continuing anyway"
-                self.logger.warning(msg)
+        # We need to start a span around this transaction becaues if we don't,
+        # the nested transaction shows up as "No instrumentation" in the
+        # enclosing transaction in the Sentry UI.
+        with sentry_sdk.start_span(name="startup", op="startup"):
+            with start_transaction(
+                name=f"{self.__class__.__name__} - startup",
+                op=f"{self.__class__.__name__} - startup",
+            ):
+                if self.options.jitter:
+                    with start_span(op="pre_login_delay"):
+                        max_delay = self.options.jitter.total_seconds()
+                        delay = self._random.uniform(0, max_delay)
+                        if not await self.pause(timedelta(seconds=delay)):
+                            return
+                await self.hub_login()
+                if not await self._client.is_lab_stopped():
+                    try:
+                        await self.delete_lab()
+                    except JupyterDeleteTimeoutError:
+                        msg = (
+                            "Unable to delete pre-existing lab, continuing"
+                            " anyway"
+                        )
+                        self.logger.warning(msg)
```

Comment:
Code has gotten really indented here as well, enough so that it's starting to distort error messages. I usually take that as a sign to refactor into helper methods.